dex-load/src/main/scala/com/wavesplatform/dex/load/TankGenerator.scala

```diff
@@ -0,0 +1,275 @@
+          else Transactions.makeTransferTx(issuer, acc.getAddress, 1, ass, 300000, "WAVES", "")
+        }
+      }
+      .flatMap(tx => {
+        println(s"\tSending transfer tx: ${mkJson(tx)}")
+        services.node.send(tx)
+      })
+
+    waitForHeightArise()
+  }
+
+  private def mkOrders(accounts: List[PrivateKeyAccount], pairs: List[AssetPair], matching: Boolean = true): List[Request] = {
+    print(s"Creating orders... ")
+    val safeAmount = settings.assets.quantity / 2 / accounts.length / 400
+
+    val orders = (0 to 399).flatMap(
+      _ =>
+        accounts.map(mkOrder(
+          _,
+          if (math.random < 0.5 || !matching) Type.BUY else Type.SELL,
+          Random.nextInt(safeAmount.toInt),
+          Random.nextInt(safeAmount.toInt),
+          pairs(Random.nextInt(pairs.length))
+        )))
+
+    println("Done")
+    Random.shuffle(orders.map(Request(Request.POST.toString, "/matcher/orderbook", Request.PLACE.toString, _))).toList
+  }
+
+  private def mkPairsAndDistribute(accounts: List[PrivateKeyAccount], pairsFile: Option[File]): List[AssetPair] = {
+    val assets =
+      if (Files.notExists(pairsFile.get.toPath)) mkAssets()
+      else readAssetPairs(pairsFile).map(p => { s"${p.getAmountAsset}-${p.getPriceAsset}" }).mkString("-").split("-").toSet.toList
+
+    val pairs =
+      if (Files.notExists(pairsFile.get.toPath)) mkAssetPairs(assets)
+      else readAssetPairs(pairsFile)
+
+    distributeAssets(accounts, assets)
+    pairs
+  }
+
+  private def mkPlaces(accounts: List[PrivateKeyAccount], requestsCount: Int, pairsFile: Option[File], matching: Boolean): List[Request] = {
+    println(s"Making requests for ${if (matching) "matching" else "placing"} ...")
+
+    mkOrders(accounts, mkPairsAndDistribute(accounts, pairsFile), matching).take(requestsCount)
+  }
+
+  private def mkPlaces(accounts: List[PrivateKeyAccount], requestsCount: Int, pairsFile: Option[File]): List[Request] =
+    mkPlaces(accounts, requestsCount, pairsFile: Option[File], matching = false)
+
+  private def mkMatching(accounts: List[PrivateKeyAccount], requestsCount: Int, pairsFile: Option[File]): List[Request] =
+    mkPlaces(accounts, requestsCount, pairsFile: Option[File], matching = true)
+
+  private def mkCancels(accounts: List[PrivateKeyAccount], requestsCount: Int): List[Request] = {
+    println("Making requests for cancelling...")
+
+    val cancels = accounts
+      .flatMap(a => {
+        Json
+          .parse(getOrderBook(a))
+          .as[List[JsValue]]
+          .map(o => {
+            val id = (o \ "id").as[String]
+            val aa = ((o \ "assetPair").as[JsValue] \ "amountAsset").validate[String] match {
+              case JsSuccess(name, _) => name
+              case _: JsError         => "WAVES"
+            }
+            val pa = ((o \ "assetPair").as[JsValue] \ "priceAsset").validate[String] match {
+              case JsSuccess(name, _) => name
+              case _: JsError         => "WAVES"
```

Comment:
`case _: JsError => "WAVES"` looks weird

Reply:
if we tried to validate `null` for amount/price asset validator will move to this "error", so in this situation, we realize that we should use "WAVES"

Reply:
Can we use `JsNull` here?